File: Core/src/main/java/jokatu/identity/Identifier.java
package jokatu.identity;

import java.util.function.Supplier;

/**.
 * An identifier provides identities.  The identities should be unique every time {@link Supplier#get() get()} is
 * called.
 * @author Steven Weston
 */
public interface Identifier<I extends Identity> extends Supplier<I> {

}

File: Core/src/main/kotlin/jokatu/identity/Identifier.kt
package jokatu.identity

import java.util.function.Supplier

/**.
 * An identifier provides identities.  The identities should be unique every time [get()][Supplier.get] is
 * called.
 * @author Steven Weston
 */
interface Identifier<I : Identity> : Supplier<I>
